logs/analyze.go

```diff
@@ -1634,38 +1638,43 @@ func classifyAndSetDetails(logLine state.LogLine, statementLine state.LogLine, d
 			numIndexScans, _ := strconv.ParseInt(parts[4], 10, 64)
 			pagesRemoved, _ := strconv.ParseInt(parts[5], 10, 64)
 			relPages, _ := strconv.ParseInt(parts[6], 10, 64)
-			pinskippedPages, _ := strconv.ParseInt(parts[7], 10, 64)
-			frozenskippedPages, _ := strconv.ParseInt(parts[8], 10, 64)
-			tuplesDeleted, _ := strconv.ParseInt(parts[9], 10, 64)
-			newRelTuples, _ := strconv.ParseInt(parts[10], 10, 64)
-			newDeadTuples, _ := strconv.ParseInt(parts[11], 10, 64)
 
-			// Parts 12 to 19 (Index scan info, I/O read/write timings) are only present on Postgres 14+ and dealt with at the end
+			// Parts 7 to 10 (scanned, pinskipped and frozenskipped pages) are version dependent and dealt with later
 
-			if parts[20] != "" { // Postgres 14+, with I/O information before buffers
-				readRatePart = parts[20]
-				writeRatePart = parts[21]
+			tuplesDeleted, _ := strconv.ParseInt(parts[11], 10, 64)
+			newRelTuples, _ := strconv.ParseInt(parts[12], 10, 64)
```

Comment:
This is following an existing pattern, so we shouldn't change it now, but I wonder if we shouldn't add a helper to panic if there's a parsing error instead of ignoring it. There should not be errors here, but given how many different matches we are juggling, I wouldn't be surprised if we get something wrong here sooner or later. It might be nice to complain loudly rather than proceed with bogus data.

Reply:
Yeah, agreed - probably not panic, but output a warning or similar.